api/server/call_logs.go

```diff
@@ -21,18 +27,22 @@ func (s *Server) handleCallLogGet(c *gin.Context) {
 		return
 	}
 
-	// TODO this API needs to change to text/plain / gzip anyway, punting
-	// optimization, but we can write this direct to the wire, too... seems like
-	// we should write some kind of writev json thing for go since we keep
-	// hitting this :(
 	var b bytes.Buffer
 	b.ReadFrom(logReader)
 
-	callObj := models.CallLog{
-		CallID:  callID,
-		AppName: appName,
-		Log:     b.String(),
+	mimeTypes, _ := c.Request.Header["Accept"]
+
+	for _, mimeType := range mimeTypes {
+		switch mimeType {
+		case "text/plain":
+			c.String(http.StatusOK, b.String())
```

Comment:
one of the nice benefits of having a plaintext api here is that we don't need to buffer and encode the log. we should only read the log into a buffer for the deprecated json path and for the new plaintext api, we can just copy the log directly to the response connection. for minio/s3 backend this will read directly from the s3 connection and copy it to the client response which means we aren't buffering it at all internally, which is pretty slick.